### DISCUSSION TITLE
API endpoint parameters

### DISCUSSION THREAD

**Discussion by dextermb** [sveltejs/kit#4451] [General]
Hey

I'm attempting to get the query-string parameters from an API endpoint, but I've noticed that they are not returned in `params` provided by an endpoint handler:

```js
// ./foo/[x].js
async function get({ params }) {
  console.log(params)

  return { status: 204 }
}
```

```
./foo/bar         -> { x: 'bar' }
./foo/bar?limit=1 -> { x: 'bar' }
```

Currently to get around this we have to go into the `url.searchParams` and use `get`:
```js
// ./foo/[x].js
async function get({ url }) {
  console.log(url.searchParams.get('limit'))

  return { status: 204 }
}
```

```
./foo/bar         -> null
./foo/bar?limit=1 -> 1
```

I'd like to see `url.searchParams` merged into `params`. MDN suggest that there's a [`getAll`](https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams/getAll) method which could make merging _easy_.

Thoughts?

If not how would I go about implementing a global-middleware so that it could be done in user land rather than by the framework?

**Comment by n-pn** [comment]
`getAll()` mean get all search params as an array.
for example a query string look like this `a=1&a=2` then `get('a')` will return `'1'` and `getAll('a')` will return `[1,2]`.

I don't think it has anything to do with this suggestion.

**Comment by mrkishi** [comment]
You could do this with a helper `params = merge(params, url)` if you want, but it doesn't sound likely to be added to SvelteKit itself.

As the developer, you have control over the params in your route definitions... But what happens when someone adds a query with the same name? Which one has priority, how would they get merged, how would you distinguish where each one came from? These are all issues that don't exist if you get them explicitly from either `url` or `params`.

**Reply by dextermb** [reply]
Fair enough, good points raised.

**Reply by mrkishi** [reply]
I forgot to mention that you could merge them on the global `handle` hook, or add a merged `params` to `stuff` from a common layout.
